hypershift/assets/ocp-bm-cluster.md

```diff
@@ -0,0 +1,292 @@
+
+~~~xml
+<ip family="ipv4" address="192.168.125.1" prefix="24">
+  <dhcp>
+    <range start="192.168.125.100" end="192.168.125.254"/>
+    <host mac="aa:bb:cc:dd:ee:ff" name="ocp-worker-0.hosted0.krnl.es" ip="192.168.125.10"/>
+    <host mac="aa:bb:cc:dd:ee:fa" name="ocp-worker-1.hosted0.krnl.es" ip="192.168.125.11"/>
+  </dhcp>
+</ip>
+~~~
+
+In order to do this, `virsh net-edit <network>` will open the $EDITOR to perform the changes, then `virsh net-destroy <network> && virsh net-start <network>` will apply those changes.
+
+> **NOTE**: `virsh net-destroy` won't destroy the network, it only stops it.
+
+## **Install the cluster**
+
+* Clone the baremetal plan repository:
+
+~~~sh
+git clone https://github.com/karmab/kcli-openshift4-baremetal.git
+cd kcli-openshift4-baremetal
+~~~
+
+* Create the parameters file
+
+~~~sh
+envsubst <<"EOF" > ocp.yml
+version: stable
+tag: "4.9"
+cluster: ocp
+domain: ${OCPCLUSTER_NETWORK_DOMAIN}
+provisioning_enable: false
+ztp_spoke_deploy: false
+virtual_masters: true
+virtual_masters_memory: 32768
+virtual_masters_baremetal_mac_prefix: aa:aa:aa:aa:aa
+launch_steps: true
+deploy_openshift: true
+baremetal_cidr: ${OCPCLUSTER_NETWORK_CIDR}
+baremetal_net: ${OCPCLUSTER_NETWORK_NAME}
+api_ip: ${API_IP}
+ingress_ip: ${INGRESS_IP}
+pullsecret: pull-secret.json
+installer_numcpus: 4
+installer_memory: 4096
+EOF
+~~~
+
+* Run the installation:
+
+~~~sh
+kcli create plan ocp --pf ocp.yml
+~~~
+
+We want to use some persistent volumes so we will deploy the Local Storage Operator and configure it to use a couple of disks on each master.
+
+* Add more disks to the masters:
+
+~~~sh
+kcli update vm ocp-master-0 -P disks=[30,10,10]
+kcli update vm ocp-master-1 -P disks=[30,10,10]
+kcli update vm ocp-master-2 -P disks=[30,10,10]
+~~~
+
+The installation progress can be followed as:
+
+~~~sh
+kcli console -s ocp-installer
+~~~
+
+While the installation finishes (it takes a while), a couple of empty VMs to be used as workers can be created:
+
+~~~sh
+kcli create vm -P machine=pc-q35-rhel8.2.0 -P disks=['{"diskname":"sda","size":"120","interface": "scsi"}'] -P uuid=11111111-1111-1111-1111-111111111111 -P start=false -P nets=['{"name":"hosted0-krnl","mac":"aa:bb:cc:dd:ee:ff"}'] -P numcpus=8 -P memory=8192 ocp-worker-0
+kcli create vm -P machine=pc-q35-rhel8.2.0 -P disks=['{"diskname":"sda","size":"120","interface": "scsi"}'] -P uuid=11111111-1111-1111-1111-111111111112 -P start=false -P nets=['{"name":"hosted0-krnl","mac":"aa:bb:cc:dd:ee:fa"}'] -P numcpus=8 -P memory=8192 ocp-worker-1
+~~~
+
+## **Verify the installation**
+
+Once finished, you can jump into the _installer_ VM where the KUBECONFIG variable is already set (as root):
+
+~~~sh
+kcli ssh ocp-installer
+
+sudo -i
+oc get clusterversion
+~~~
+
+## **Local storage Operator**
+
+* Create the namespace
+
+~~~sh
+oc create namespace openshift-local-storage
```

Comment:
Use tasty for consistency with other docs?

Reply:
In this case I wanted to show how to do it in the 'traditional' way :) I'll add how to do it with `tasty` as well.